src/topp/DatabaseSuitability.cpp

```diff
@@ -196,153 +189,38 @@ class DatabaseSuitability :
-        }
-
-        sort(hits.begin(), hits.end(),
-          [](const PeptideHit& a, const PeptideHit& b)
-          {
-            return float(a.getMetaValue("q-value")) < float(b.getMetaValue("q-value"));
-          });
-      }
-
-
-      const PeptideHit& top_hit = hits[0];
-
-      // skip if the top hit is a decoy hit
-      if (!top_hit.metaValueExists("target_decoy"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy information found! Make sure 'PeptideIndexer' is run beforehand."));
-      }
-      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
-
-      // skip if top hit is out ouf FDR
-      if (scoreHigherThanFDR_(top_hit, FDR, q_value_score)) continue;
-
-      // check if top hit is found in de novo protein
-      if (!isNovoHit_(top_hit)) // top hit is db hit
-      {
-        ++count_db;
-        continue;
-      }
-
-      // find the second target hit, skip all decoy or novo hits inbetween
-      const PeptideHit* second_hit = nullptr;
-      String target = "target";
-      for (UInt i = 1; i < hits.size(); ++i)
-        {
-        // check for FDR
-        if (scoreHigherThanFDR_(hits[i], FDR, q_value_score)) break;
-
-        if (target.find(String(hits[i].getMetaValue("target_decoy"), 0)) == 0) // also check for "target+decoy" value
-        {
-          // check if hit is novo hit
-          if (isNovoHit_(hits[i])) continue;
-          
-          second_hit = &hits[i];
-          break;
-        }
-      }
-      if (second_hit == nullptr) // no second target hit with given FDR found
-      {
-        ++count_novo;
-        continue;
-      }
-
-      // second hit is db hit
-      ++count_interest;
-
-      // check for re-ranking
-      if (no_re_rank)
-      {
-        ++count_novo;
-        continue;
-      }
-
-      // check for xcorr score
-      if (!top_hit.metaValueExists("MS:1002252") || !second_hit->metaValueExists("MS:1002252"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No cross correlation score found at peptide hit. Only Comet search engine is supported right now."));
-      }
-      
-      double top_xscore_mw = double(top_hit.getMetaValue("MS:1002252")) / top_hit.getSequence().getMonoWeight();
-      double second_xscore_mw = double(second_hit->getMetaValue("MS:1002252")) / second_hit->getSequence().getMonoWeight();
-      if (top_xscore_mw - second_xscore_mw <= cut_off)
-      {
-        ++count_db;
-        ++count_re_ranked;
-      }
-      else
-      {
-        ++count_novo;
-      }
-    }
-
-    double suitability = double(count_db) / (count_db + count_novo); //db suitability
-
-    // spectra quality
-
-    Size count_novo_seq = 0;
+    Ms2IdentificationRate q;
+    q.compute(novo_peps, exp, true);
```

Comment:
add a sanity check that all PepIDs in `novo_peps` are actually deNovo protein hits?
Otherwise the user can accidentally pass the wrong idXML and hell will break loose?